File: problems/memento/attachments_source/cpp/memento.cpp
#include <vector>
#include <cassert>
#include "memento.h"

std::vector<std::pair<int,int>> run(std::vector<std::pair<int,int>> edges)
{
    int M=edges.size(); // number of edges
    bool first_time = true; // TODO: determine if first time

    if (first_time)
    {
        std::vector<std::pair<int,int>> extra_edges = {};
        extra_edges.emplace_back(0,1);
        // TODO: add extra edges
        assert(extra_edges.size()==30 && "Add more edges to the graph!");
        return extra_edges;
    }
    else
    {
        return {};
    }
}

File: problems/memento/attachments_source/memento/python3/memento.py
#!/usr/bin/python3
def run(edges):
    M=len(edges) # number of edges
    first_time = True # TODO: determine if first time

    if first_time:
        extra_edges = []
        extra_edges.append([0,1])
        # TODO: add extra edges
        assert len(extra_edges)==30, "Add more edges to the graph!"
        return extra_edges
    else:
        return []
